### ISSUE TITLE
Launching jupyter lab opens an empty browser page

### ISSUE THREAD

**Issue by eco32i** [jupyter/jupyterlab#408]
Both in Chrome and Firefox. Ubuntu 16.04, pip-installed in virtualenv:

```
(pydata3) ilya@barmalei:~$ jupyter lab
[W 16:29:24.588 LabApp] Unrecognized JSON config file version, assuming version 1
[I 16:29:24.877 LabApp] Pre-alpha version of JupyterLab extension loaded from /home/ilya/.venv/pydata3/lib/python3.5/site-packages/jupyterlab
[I 16:29:24.879 LabApp] Serving notebooks from local directory: /home/ilya
[I 16:29:24.879 LabApp] 0 active kernels 
[I 16:29:24.879 LabApp] The Jupyter Notebook is running at: http://localhost:8888/
[I 16:29:24.879 LabApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
Created new window in existing browser session.
[W 16:31:45.780 LabApp] 404 GET /favicon.ico (127.0.0.1) 17.08ms referer=None
[W 16:31:45.797 LabApp] 404 GET /favicon.ico (127.0.0.1) 1.18ms referer=None
```

browser console:

```
bundle.js:12586 Uncaught (in promise) Error: No registered provider for type: DocumentRegistry. Provider resolution path: .
    at missingProviderError (http://localhost:8888/lab/bundle.js:12586:17)
    at Object.resolveService (http://localhost:8888/lab/bundle.js:12535:36)
    at ServiceRegistry.resolveService (http://localhost:8888/lab/bundle.js:12441:25)
    at http://localhost:8888/lab/bundle.js:12284:75
    at Array.map (native)
    at ExtensionRegistry.activateExtension (http://localhost:8888/lab/bundle.js:12284:38)
    at Application.activateExtension (http://localhost:8888/lab/bundle.js:362:34)
    at http://localhost:8888/lab/bundle.js:431:73
    at Array.map (native)
    at http://localhost:8888/lab/bundle.js:431:40
```

**Comment by sccolbert** [comment]
Thanks! This is a known issue that we are actively working on at the moment.

CC @blink1073 @afshin

**Comment by sccolbert** [comment]
Update: For some reason DocumentRegistry is getting bundled twice, and the version which is getting registered as a service provider is *not* the one getting imported by the things requesting the instance. This is causing the failed registry lookup. Stay tuned.

**Comment by sccolbert** [comment]
Looks like a version clash in ipywidgets is causing jupyterlab to be bundled twice. We're working on a fix.

**Comment by elchinot7** [comment]
Same problem here using pip-installed `jupyterlab` ! 

After launching `jupyter lab` I got the `ERROR:nss_util.cc(839) `

    [W 17:07:45.427 LabApp] Unrecognized JSON config file version, assuming version 1
    [I 17:07:45.637 LabApp] Pre-alpha version of JupyterLab extension loaded from /usr/local/lib/python2.7/dist-packages/jupyterlab
    [I 17:07:45.640 LabApp] Serving notebooks from local directory: /home/chinot7
    [I 17:07:45.640 LabApp] 0 active kernels 
    [I 17:07:45.640 LabApp] The Jupyter Notebook is running at: http://localhost:8888/
    [I 17:07:45.640 LabApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
    [4153:4184:0715/170746:ERROR:nss_util.cc(839)] After loading Root Certs, loaded==false: NSS error code: -8018
    Created new window in existing browser session.
    [I 17:10:22.646 LabApp] 302 GET / (::1) 0.88ms

**Comment by jasongrout** [comment]
Thanks for reporting this - totally my fault. We fixed it on master, and I'm waiting on merging one more PR before releasing new packages.

**Comment by jasongrout** [comment]
Release 0.1.1 fixes this. Thanks again for reporting it!

**Comment by drussellmrichie** [comment]
I did "pip install jupyterlab --upgrade" at the terminal, "conda list" at the terminal confirmed that I have jupyterlab 0.1.1, and I'm still getting this error...? This is what my terminal spits out when I enter "jupyter lab":

```
russellrichie@Russells-MacBook-Pro:~$ jupyter lab
[W 01:18:29.600 LabApp] Unrecognized JSON config file version, assuming version 1
[W 01:18:29.602 LabApp] Unrecognized JSON config file version, assuming version 1
[I 01:18:30.253 LabApp] [nb_conda_kernels] enabled, 2 kernels found
[I 01:18:30.315 LabApp] Pre-alpha version of JupyterLab extension loaded from //anaconda/lib/python3.5/site-packages/jupyterlab
[I 01:18:30.321 LabApp] Serving notebooks from local directory: /Users/russellrichie
[I 01:18:30.322 LabApp] 0 active kernels 
[I 01:18:30.322 LabApp] The Jupyter Notebook is running at: http://localhost:8888/
[I 01:18:30.322 LabApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
```

Happens in Chrome. Mac OSX El Capitan 10.11.5.

**Comment by jasongrout** [comment]
Can you try visiting the jupyterlab URL in incognito mode to make sure it's not your browser caching your javascript?

**Comment by eco32i** [comment]
Works for me. Thanks!!!

**Comment by benmarwick** [comment]
I'm on Windows 7 with Chrome 51.0.2704.103 m

Here's what I see in the terminal:

```
$ conda list
# packages in environment at C:\Anaconda3:
# 
...
jupyter                   1.0.0                    py35_3
jupyter_client            4.3.0                    py35_0
jupyter_console           4.1.1                    py35_0
jupyter_core              4.1.0                    py35_0
jupyterlab                0.1.1                     <pip>
...

$ pip install jupyterlab --upgrade
Requirement already up-to-date: jupyterlab in c:\anaconda3\lib\site-packages
Requirement already up-to-date: notebook>=4.2.0 in c:\anaconda3\lib\site-packages (from jupyterlab)

$ jupyter notebook --version
4.2.1

$ jupyter lab --version
4.2.1

$ jupyter lab
[W 00:21:33.647 LabApp] Unrecognized JSON config file version, assuming version                   1
[W 00:21:33.650 LabApp] Unrecognized JSON config file version, assuming version                   1
[I 00:21:34.190 LabApp] [nb_conda_kernels] enabled, 1 kernels found
[I 00:21:34.263 LabApp] Pre-alpha version of JupyterLab extension loaded from c:                  \anaconda3\lib\site-packages\jupyterlab
[I 00:21:34.777 LabApp] Serving notebooks from local directory: C:\emacs
[I 00:21:34.777 LabApp] 0 active kernels
[I 00:21:34.777 LabApp] The Jupyter Notebook is running at: http://localhost:888                  8/
[I 00:21:34.777 LabApp] Use Control-C to stop this server and shut down all kern                  els (twice to skip confirmation).

```

Then I get a blank browser window. Here's what I see in the browser console:

```
jquery-2.1.0.min.js:2 Uncaught TypeError: Cannot read property 'slice' of null
kernel.js:210 Uncaught (in promise) Error: Invalid KernelSpecs Model  at http://localhost:8888/lab/bundle.js:19335:20
```

And in incognito mode:

```
bundle.js:19335 Uncaught (in promise) Error: Invalid KernelSpecs Model(…)
```

**Comment by sccolbert** [comment]
@benmarwick That might be a duplicate of #417

**Comment by drussellmrichie** [comment]
I'm getting console messages similar but perhaps not quite identical to those that @benmarwick is getting (and I'm also running jupyter notebook 4.2.1 -- I didn't mention this before). In a regular tab:

`bundle.js:19335 Uncaught (in promise) Error: Invalid KernelSpecs Model(…)`

In incognito mode, the very same thing:

`bundle.js:19335 Uncaught (in promise) Error: Invalid KernelSpecs Model(…)`

Using chrome Version 51.0.2704.106 (64-bit).

Sorry for the hassle. Looking forward to being able to try out Jupyter Lab!

**Comment by sccolbert** [comment]
@drussellmrichie this is being tracked in #417

**Comment by drussellmrichie** [comment]
Gotcha, @sccolbert. Thanks. I'll watch that issue!

**Comment by forwardcrv** [comment]
Via release 0.1.1, I'm getting 404 errors on both the the mac terminal and in chrome(even in incognito)
This is what I see:
404 GET /lab (::1) 44.92ms referer=None

**Comment by benmarwick** [comment]
@sccolbert Thanks for that helpful suggestion. I cloned the repo and built v0.1.1 as suggested in #417 and I can now see it perfectly in my browser! I also [installed the R kernel](https://github.com/IRkernel/IRkernel) with no problems, thanks to the excellent instructions at that repo.  Really delighted to see this:

![bzbsi](https://cloud.githubusercontent.com/assets/1262179/16901062/88f16a84-4c7b-11e6-9803-e1632218f04e.png)

**Comment by Feynman27** [comment]
Okay. So I got it working. Here's what worked for me.

1.  git clone https://github.com/jupyter/jupyterlab.git
2. cd jupyterlab
3. Install node.js from https://nodejs.org/en/
4. npm install
5. npm run build:all
6. pip install -e .
7. jupyter serverextension enable --py jupyterlab
8. jupyter lab

I really don't think all these steps were necessary, but something worked. Hope it helps others out there.

**Comment by drussellmrichie** [comment]
I finally got this working. I followed @Feynman27's instructions, downloading node from nodejs.org instead of using brew. There also seemed to be devilish permissions-related errors and warnings, which I had to circumvent with 'sudo' and '--unsafe-perm' (the latter for npm commands). I also still seemed to get some warnings at the npm step (steps 4 and 5) regardless of this circumvention. But, pip install proceeded normally (in contrast to previous attempts), and jupyter lab appears in my browser, seemingly like it should.

Like @Feynman27, I'm not entirely sure what was the missing link from before, but something indeed did the trick.

Excited to try this out!

**Comment by jasongrout** [comment]
> pip install -e .

This should run the `npm install` and `npm run build:all` commands.

**Comment by jasongrout** [comment]
Oh, and @Feynman27, if you're just installing it to *use* it (as opposed to development), it should be enough to `pip install jupyterlab`, then `jupyter serverextension enable --py --sys-prefix jupyterlab`.

**Comment by Feynman27** [comment]
@jasongrout I agree, but when trying it with `pip install jupyterlab` followed by `jupyter serverextension enable --py --sys-prefix` `jupyterlab`, I got a blank browser. Still unclear to me why.

**Comment by damianavila** [comment]
> @drussellmrichie this is being tracked in #417

This is not directly impacting jupyterlab now, but just to let you know, we've released versions of nb_conda, nb_conda_kernels and nb_anacondacloud through the official Anaconda conda repos that should resolve the issues you saw with Python[root].

**Comment by thomasaarholt** [comment]
I just want to point out, for others coming this way for the white-window problem - @Feynman27's solution solved it for me. I was upgrading from a conda install.

It might be that simply doing `pip install -e .` is enough, but at least following the procedure fixed my problem.

**Comment by sizhky** [comment]
I can't make it work. When I run ```pip install -e .``` it says ```Obtaining file:///C:/tmp/jupyterlab``` and is stuck there. I've waited for an hour and it's still there. I don't know what the issue is.  
And even before that, I was unable to run ```npm run build:all```. Instead I ran ```npm run build```.

**Comment by blink1073** [comment]
Hi @sizhky, can you please try `git clean -dfx` and then run the commands again?  Also, there is no `npm run build:all` script, `npm run build:main` is the one you're looking for.
